gnmi_server/gnoi.go

```diff
@@ -158,18 +172,29 @@ func (srv *SystemServer) Reboot(ctx context.Context, req *gnoi_system_pb.RebootR
 	}
 	log.V(1).Info("gNOI: Reboot")
 	log.V(1).Info("Request:", req)
-	log.V(1).Info("Reboot system now, delay is ignored...")
-	// TODO: Support GNOI reboot delay
-	// Delay in nanoseconds before issuing reboot.
-	// https://github.com/openconfig/gnoi/blob/master/system/system.proto#L102-L115
-	config_db_json, err := io.ReadFile(fileName)
-	if errors.Is(err, os.ErrNotExist) {
-		fileName = ""
-	}
-	err = RebootSystem(string(config_db_json))
-	if err != nil {
-		return nil, err
+
+	// Check the reboot type
+	if req.GetMethod() == gnoi_system_pb.RebootMethod_HALT {
```

Comment:
Maybe using switch statement here to make it easier to add more method?

Reply:
Done. Thanks!